Undefined behaviour in `bool TimeMap::isTimestepInFreqSequence()`

The member function
```C++
bool TimeMap::isTimestepInFreqSequence(size_t timestep, size_t start_timestep, size_t frequency, bool years) const
```
which is defined in [`opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp`](https://github.com/OPM/opm-parser/blob/master/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp#L251) contains the following statement on [line 274](https://github.com/OPM/opm-parser/blob/master/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp#L274)
```C++
int dist = std::distance( ci_start_timestep - 1, ci_timestep );
```
with `ci_start_timestep` and `ci_timestep` both being positions (iterators) into a collection, `timesteps`.

The above statement is undefined if `ci_start_timestep` happens to be equal to `timesteps.begin()`, because we are generally not allowed to form the iterator `begin() - 1` of any collection.  The test `runIOConfigIntegrationTest` triggers this undefined behaviour and libstdc++'s Debug Mode produces a diagnostic of the form
```
/usr/include/c++/4.6/debug/safe_iterator.h:307:error: attempt to retreat a 
    dereferenceable (start-of-sequence) iterator 1 steps, which falls     
    outside its valid range.
```

I don't know what the correct solution is in this case because I don't know the code in question.  I only know that this issue may cause problems in the future.

--- a/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp
+++ b/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp
@@ -271,7 +271,7 @@ namespace Opm {
         if (start_ts_in_timesteps) {
             //Pick every n'th element, starting on start_timestep + (n-1), that is, every n'th element from ci_start_timestep - 1 for freq n > 1
             if (ci_timestep >= ci_start_timestep) {
-                int dist = std::distance( ci_start_timestep - 1, ci_timestep );
+                int dist = std::distance( ci_start_timestep, ci_timestep ) + 1;
                 if ((dist % frequency) == 0) {
                     timestep_right_frequency = true;
                 }

